### PR TITLE
Delete an application member

### PR DESCRIPTION
**Pull request by drummond-dds**
* [Pivotal Tracker](https://www.pivotaltracker.com/story/show/165062523)

```gherkin
Scenario: 
Given I am logged in as a user with Edit Team privs on an application and looking at the Team settings page
When I click Delete Member,
Then I see a modal per the design,
And the modal has a warning banner reading "Warning! This action is permanent. {Username} will no longer be able to access this application,"
And there is text reading "Please type the word 'DELETE' to confirm:",
And there is a text box,
And there is a Delete User button that is grayed out until/unless the text box contains 'delete' (case insensitive),
And there is a cancel button,
Then I click Delete User,
Then I am taken back to the page I was on, to the table I was at, with a success banner - "You have successfully deleted {username} from {Application-name}"
And the user is set to No Access for all environments in the application.
```

### CODE REVIEW

File: atst/domain/applications.py [hunk 1]
```diff
@@ -100,3 +101,15 @@ def create_member(
...
 
         return application_role
+
+    @classmethod
+    def remove_member(cls, application, user):
+        application_role = ApplicationRoles.get(
+            user_id=user.id, application_id=application.id
+        )
+
+        db.session.delete(application_role)
```
Comment (dandds) [inline atst/domain/applications.py]: I'm not sure about the answer to this question, but we should talk through whether this should actually delete the role or set it to disabled: https://github.com/dod-ccpo/atst/blob/master/atst/models/application_role.py#L15

File: atst/routes/applications/team.py [hunk 2]
```diff
@@ -158,3 +158,31 @@ def create_member(application_id):
             _anchor="application-members",
         )
     )
+
+
+@applications_bp.route(
+    "/applications/<application_id>/members/<user_id>/delete", methods=["POST"]
+)
+# TODO: Is this correct??
+@user_can(Permissions.EDIT_APPLICATION_MEMBER, message="remove application member")
```
Comment (dandds) [inline atst/routes/applications/team.py]: You mean, is EDIT_APPLICATION_MEMBER the correct permission? Yes. We could add DELETE_APPLICATION_MEMBER in the same permission set for granularity, which might be a good idea.

File: atst/routes/applications/team.py [hunk 3]
```diff
@@ -158,3 +158,31 @@ def create_member(application_id):
...
+def remove_member(application_id, user_id):
+    application_role = ApplicationRoles.get(
+        application_id=application_id, user_id=user_id
+    )
+
+    Applications.remove_member(application=g.application, user=application_role.user)
```
Comment (dandds) [inline atst/routes/applications/team.py]: `remove_member` also does `ApplicationRoles.get`. I don't think we should do it in both places.

File: tests/domain/test_applications.py [hunk 4]
```diff
@@ -155,3 +158,25 @@ def test_for_user():
...
+    )
+
+    Applications.remove_member(application=application, user=member_role.user)
+
+    with pytest.raises(NotFoundError):
+        ApplicationRoles.get(user_id=user.id, application_id=application.id)
+
+    #
+    # TODO: Why does above raise NotFoundError and this returns None
+    #
```
Comment (dandds) [inline tests/domain/test_applications.py]: I don't know the original rationale for `EnvironmentRoles.get` returning one or `None`. If you can change it to `raise` without breaking anything, please do. (This brings up the whole "don't commit until you're really done making changes" question again, but we can punt on that a while longer.)

File: templates/portfolios/applications/team.html [hunk 5]
```diff
@@ -109,6 +112,35 @@
...
+          {% set delete_modal_id = "delete-user-{}".format(member_form.id) %}
+          {% call Modal(name=delete_modal_id, dismissable=True) %}
+            <h1>
+              {{ "portfolios.applications.remove_member.header" | translate }}
+            </h1>
+
+            {{
+              Alert(
+                title=("portfolios.applications.remove_member.alert.title" | translate),
+                message=("portfolios.applications.remove_member.alert.message" | translate({"user_name": member_form.user_name.data})),
```
Comment (dandds) [inline templates/portfolios/applications/team.html]: The modal still has application-specific language in it.

File: translations.yaml [hunk 6]
```diff
@@ -408,6 +408,12 @@ portfolios:
     app_settings_text: App settings
     create_button_text: Create
     csp_console_text: CSP console
+    remove_member:
+      alert:
+        message: '{user_name} will no longer be able to access this application'
+        title: Warning! This action is permanent.
```
Comment (dandds) [inline translations.yaml]: This is in at least three places now. Can we consolidate?

File: atst/routes/applications/team.py (hunk 3 above)
Comment (dandds) [reply atst/routes/applications/team.py]: I guess you could argue that the domain method should be self-contained. At a minimum, could you confirm that SQLAlchemy's basic ID cache is working here and we don't issue two equivalent SELECT statements?

File: tests/domain/test_applications.py (hunk 4 above)
Comment (drummond-dds) [reply tests/domain/test_applications.py]: I just did a quick spike and I think we should defer this to a future follow up PR

File: atst/routes/applications/team.py (hunk 2 above)
Comment (drummond-dds) [reply atst/routes/applications/team.py]: Added in f1299dc4b0e22af8f07714dbb56c9f9009a652f8

File: translations.yaml (hunk 6 above)
Comment (drummond-dds) [reply translations.yaml]: Fixed in b884c8a762eb2ad92476345513aafd35babcbdcb

File: templates/portfolios/applications/team.html (hunk 5 above)
Comment (drummond-dds) [reply templates/portfolios/applications/team.html]: I think this is in the requirements

```cucumber
And the modal has a warning banner reading "Warning! This action is permanent. {Username} will no longer be able to access this application,"
```

Comment (drummond-dds) [reply templates/portfolios/applications/team.html]: Screen Shot 2019-05-14 at 15 13 56

File: atst/domain/applications.py (hunk 1 above)
Comment (drummond-dds) [reply atst/domain/applications.py]: Fixed in 9927b22783e2deaac19e239fe068cd5e6e3d22e5

**Pull request merged**